src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -185,18 +185,18 @@ describe('ServerMetricsCollector', () => {
     let metrics = await collector.collect();
     expect(metrics.concurrent_connections).toEqual(0);
 
-    sendGet('/').end(() => null);
+    const res1 = sendGet('/').then(res => res);
```

Comment:
Supertest API... The request is sent when calling/awaiting `.then`. I'm forced to noop-chain the promise to execute the request without awaiting it.

Reply:
How it's different from `.end()`? I noticed that in case of `end()`, it logs `superagent request was sent twice, because both .end() and .then() were called. Never call .end() if you use promises`. Does it send a request twice 🤦

Reply:
Would you mind adding a comment about the usage of such a sophisticated constriction in the test?

Reply:
> Does it send a request twice

Yea...

```ts
const req = sendGet('/').end(() => null);
// later
await req;
```

Does send the request on each line... Imho having `.then` executing the request is a very bad decision from this lib, but that's how it works... Which is why I had to use this `.then` trick.

Will add a comment